### ISSUE TITLE
[question] input combos for example strg+c

### ISSUE THREAD

**Issue by christopherpross** [Apostolique/Apos.Input#5]
Hey,
nice project!
I would to ask If you support key presses there are pressed at the same time, not only with modifiers, with all keys and if possible with mouse and keyboard combos.
This would be very great. 

All the best,

**Comment by Apostolique** [comment]
There's no difference between modifiers and other keys from this library's point of view.

Let's say you want to trigger when D + F + Mouse Left click, you can do:

```csharp
ConditionComposite coolCombo =
    new ConditionComposite(
        new ConditionSet(
            new ConditionKeyboard(Keys.D),
            new ConditionKeyboard(Keys.F),
            new ConditionMouse(MouseButton.LeftButton)
        )
    );

if (coolCombo.Pressed()) {
    // Do something.
}
```

**Comment by christopherpross** [comment]
Hey,
thank you for your quick answer,
only one other question regarding this topic.
giving following example:

- uparrow move forward 
- shift+uparrow run forward
- strg+uparrow sneak forward

what happen now if I press strg+shift+uparrow?
Can I, without define on each combo many not to use keys, forbit such collisions between combos? The easiest way, is to check the count of the keys array in the keyboardstate for example. Exist there a better way?
I hope you anderstand my question.

**Comment by Apostolique** [comment]
One way is to use an if, else if chain:

```csharp
ConditionComposite forward =
    new ConditionComposite(
        new ConditionSet(new ConditionKeyboard(Keys.Up))
    );

ConditionComposite runForward =
    new ConditionComposite(
        new ConditionSet(
            new ConditionKeyboard(Keys.LeftShift),
            new ConditionKeyboard(Keys.Up)
        )
    );

ConditionComposite sneakForward =
    new ConditionComposite(
        new ConditionSet(
            new ConditionKeyboard(Keys.LeftControl),
            new ConditionKeyboard(Keys.Up)
        )
    );

if (sneakForward.Held()) {
    // Sneak
} else if (runForward.Held()) {
    // Run forward
} else if (forward.Held()) {
    // Move forward
}
```

____

I am still thinking about this subject in #1. My current idea is to introduce a token concept. You could bind a token to a composite, set or condition. If that token is triggered, other conditions with the same token would not be triggered. But I want to think about this more before I add it to the API. What are your thoughts on this token idea?

**Comment by Apostolique** [comment]
@christopherpross I have just released version 2.0. In that version, there is no longer a need for the "not keys". Also since your last message, the API has been cleaned up.

Here is how you would code your previous example:

```csharp
using Apos.Input;
using Track = Apos.Input.Track;

ICondition forward = new Track.KeyboardCondition(Keys.Up);

ICondition runForward =
    new AllCondition(
        new KeyboardCondition(Keys.LeftShift),
        new Track.KeyboardCondition(Keys.Up)
    );

ICondition sneakForward =
    new AllCondition(
        new KeyboardCondition(Keys.LeftControl),
        new Track.KeyboardCondition(Keys.Up)
    );

if (sneakForward.Held()) {
    // Sneak
}
if (runForward.Held()) {
    // Run forward
}
if (forward.Held()) {
    // Move forward
}
```

You have to put the condition to prioritize as the first check. You also prefix the keys that should be unique with `Track.`.

With that, I'll consider this issue solved.
